packages/store/src/usersSlice/thunks.ts

```diff
@@ -535,22 +638,22 @@ const transformAreaRoleDataToScopedPermissionsSets = (area, areaPermissions, exc
       return itemPermissionAccu;
     }, accu);
     return accu;
-  }, {});
+  }, {} as PermissionSetObject);
   return Object.entries(permissionSetObject).map(([name, { value, ...scope }]) => {
     if (value.includes(excessiveAccessSelector)) {
-      return { name };
+      return { name: name as PermissionSetIds };
     }
-    return { name, scope: { ...scope, value: value.filter(duplicateFilter) } };
+    return { name: name as PermissionSetIds, scope: { ...scope, value: value.filter(duplicateFilter) } };
   });
 };
 
-const transformRoleDataToRole = (roleData, roleState = {}) => {
+const transformRoleDataToRole = (roleData: SubmittedRole, roleState: Partial<Role> = {}): { permissionSetsWithScope: PermissionSetWithScope[]; role: Role } => {
   const role = { ...roleState, ...roleData };
   const { description = '', name, uiPermissions = emptyUiPermissions } = role;
   const { maybeUiPermissions, remainderKeys } = Object.entries(emptyUiPermissions).reduce(
-    (accu, [key, emptyPermissions]) => {
+    (accu: { maybeUiPermissions: any; remainderKeys: UiPermissionsByAreaKeys[] }, [key, emptyPermissions]) => {
```

Comment:
would the `maybeUiPermissions` not also be `{ item: string; uiPermissions: UiPermissionsByIdKey[] }[]`?

Reply:
no, it would be `Partial<Record<ScopedPermissionsByAreaKey, Record<string, AnyPermission[]>>>`

Reply:
so not `any`.